opensearch-aws-sigv4/lib/opensearch-aws-sigv4.rb

```diff
@@ -45,24 +45,30 @@ def initialize(transport_args = {}, sigv4_signer, &block)
         end
 
         @sigv4_signer = sigv4_signer
-        super transport_args, &block
+        super(transport_args, &block)
       end
 
       # @see OpenSearch::Transport::Transport::Base::perform_request
       def perform_request(method, path, params = {}, body = nil, headers = nil)
-        verify_open_search unless @verified
         signature_body = body.is_a?(Hash) ? body.to_json : body.to_s
         signature = sigv4_signer.sign_request(
           http_method: method,
           url: signature_url(path, params),
           headers: headers,
           body: signature_body)
         headers = (headers || {}).merge(signature.headers)
-        @transport.perform_request(method, path, params, body, headers)
+        super(method, path, params, body, headers)
       end
 
       private
 
+      def verify_open_search(headers = nil)
+        verify_signature = sigv4_signer.sign_request(
+          http_method: 'GET',
+          url: signature_url('/', {}))
+        super(verify_signature.headers)
```

Comment:
Wouldn't this error out because `OpenSearch::Client#verify_open_search`, the super method of this one, doesn't accept any params?

I think the only change needed is overriding `open_search_validation_request` method:
```ruby
def open_search_validation_request
  signature = sigv4_signer.sign_request(http_method: 'GET', url: signature_url('/', {}))
  @transport.perform_request('GET', '/', {}, nil, signature.headers)
end
```